src/test/scala/cognite/spark/v1/DataModelInstancesRelationTest.scala

```diff
@@ -1080,9 +1080,36 @@ class DataModelInstancesRelationTest
         props.get("prop_direct_relation").map(_.value) shouldBe None
       }
     )
+
+    tryTestAndCleanUp(
+      Seq(randomId), {
+        retryWhile[Boolean](
```

Comment:
I don't understand the logic behind this retry function.
If there's an issue, it should fail fast and we'll know there's an issue.
I think this retry just hide errors.

Reply:
it helps to retry (max 20  times with backoff delay) and is used in many places of Spark DS
https://github.com/search?l=Scala&q=org%3Acognitedata+retryWhile&type=Code
When the api is not stable enough I think it's just fine to retry it anyway, we can reduce the number of retries though